### PR TITLE
Automatic mode for 'exchange-capabilities'

### PR DESCRIPTION
**Pull request by ttodua**
As @frosty00 suggested https://github.com/ccxt/ccxt/pull/10877#issuecomment-992796740 , I'll for now just submit a potential version of 'capabilities' file to automatically detect the functionalities.  later, frosty will make it toward perfection whenever he has time.

### CODE REVIEW

**Comment by ttodua** [comment]
Seems this PR doens't correctly handle the problem. For example, 'fetchBidsAsks' method is defned in base-class (even though it just throws 'not implemented' exception) is shown as green.

**Review by ttodua — COMMENTED** [review]
This handles inherited & self definition issues

**Comment by frosty00** [comment]
@ttodua yep, basically what i was thinking was to define the `.has` array dynamically from the methods inside of each exchange, will need to parse the actual file probably. However i don't think it is completely necessary to implement this inside of exchange-capabilities

**Comment by frosty00** [comment]
so basically it would be code generator method similar to our existing ones

**Comment by frosty00** [comment]
something like this - https://github.com/ccxt/ccxt/blob/master/build/inject-fields.js

**Comment by ttodua** [comment]
@frosty00 yes, for now, just for my (and probably for @samgermain ) exchange-capabilities is a very useful information for planning the tasks and getting the correct visuals about capabilities, so before you'll implement that solution, temporarily this PR will help us.

**Comment by ttodua** [comment]
@kroitor and might be ready this too, for now, till any better solution implemented.  Lmk if you find any bug. You can view the results also if you test it out.

File: examples/js/exchange-capabilities.js [hunk 1]
```diff
@@ -20,15 +20,32 @@ const isWindows = process.platform == 'win32' // fix for windows, as it doesn't
     let inexistentApi = 0
     let implemented = 0
     let emulated = 0
+
```
Comment (samgermain) [inline examples/js/exchange-capabilities.js:23]: Can you remove this

**Comment by samgermain** [comment]
> potential version of 'capabilities' file to automatically detect the functionalities

I don't understand how this PR does that

----------

Also, why is `notImplemented` dark red now? The point of having it as light red before was that it was more noticeable, and the dark red ones were the ones that couldn't be implemented, so they kind of blended into the background. That's also why the text at the bottom is bright red instead of dark red

File: examples/js/exchange-capabilities.js (hunk 1 above)
Comment (kroitor) [reply examples/js/exchange-capabilities.js:23]: Which specific code to remove in this PR do you mean? All of it?

**Comment by ttodua** [comment]
@samgermain about colors - my mistake. I've swapped dark/brighter reds back as it was.

**Comment by samgermain** [comment]
> @samgermain about colors - my mistake. As I am on Windows (and many console colors doesn't work in PowerShell) I had changed them temporarily. however, at this moment I've swapped dark/bright reds back as it was.

Do you know about `cmder`?

**Comment by ttodua** [comment]
@samgermain  thanks for reference.  no, I've never used that. I only use PS/CMD for shell commands on windows. If the workload will necessitate to use 3rd party tools for that, I'll might give it a shot. However, in most cases, I refrain to use 3rd party (coming from unknown sources) tool on my Win. (cmder seems ok, turns out it's open sourced on github).

**Comment by ttodua** [comment]
@kroitor this appveyor fails too.

**Comment by ttodua** [comment]
This PR has issue, I'm working on fixing.

**Comment by ttodua** [comment]
@kroitor conflicts resolved. I think this is ready now.

**Comment by kroitor** [comment]
```
~/ccxt $ git checkout -b ttodua-patch-2 master
git pull https://github.com/ttodua/ccxt.git patch-2

Switched to a new branch 'ttodua-patch-2'
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (32/32), done.
remote: Total 40 (delta 32), reused 32 (delta 32), pack-reused 8
Unpacking objects: 100% (40/40), 9.98 KiB | 378.00 KiB/s, done.
From https://github.com/ttodua/ccxt
 * branch                  patch-2    -> FETCH_HEAD
Merge made by the 'recursive' strategy.
 examples/js/exchange-capabilities.js | 59 +++++++++++++++++++++++++++++++++++++++++++++--------------
 1 file changed, 45 insertions(+), 14 deletions(-)
~/ccxt $ node examples/js/exchange-capabilities.js
/Users/igorkroitor/ccxt/examples/js/exchange-capabilities.js:47
			'margin',
			^^^^^^^
SyntaxError: Unexpected string
```

I will rewrite this for a proper automatic detection.

**Pull request merged**